code/datums/callback.dm

```diff
@@ -68,6 +68,18 @@
 	if(usr)
 		user = WEAKREF(usr)
 
+/**
+ * Qdel a callback datum
+ * callback datums are structs and qdeleting them acts like qdeleting a list, triggering a forceful hard delete always.
+ *
+ * Arguments
+ * * force set to true to force the deletion to be allowed.
+ * * ... an optional list of extra arguments to pass to the proc
+ */
+/datum/callback/Destroy(force=FALSE, ...) 
+	..()
+	return QDEL_HINT_HARDDEL_NOW 
```

Comment:
why not `QDEL_HINT_LETMELIVE` and a stack trace (or even just the hint)
callbacks getting qdeled is semi inevitable, tho I agree we should avoid it as best we can.
Maybe stacktrace if it happens in testing, so we can nuke obvious cases of it while not spamming runtimes on live?

Reply:
whoever did that would need to fix the cases tho, which might be a bit of a headache

Reply:
> why not QDEL_HINT_LETMELIVE and a stack trace (or even just the hint)

first version of this pr had a stack trace. a fuck ton of lines were printed during unit test.

you can go review it's fail logs to see how bad it would be.

Reply:
im thinking we should make a new top level type. `/struct` and we can move that behavior to them while making it seem like less of a eggshell to find out certain datums can't be deleted.

Reply:
> > why not QDEL_HINT_LETMELIVE and a stack trace (or even just the hint)
> 
> first version of this pr had a stack trace. a fuck ton of lines were printed during unit test.
> 
> you can go review it's fail logs to see how bad it would be.

I did create and destroy you are like tiny baby (I may just do this as a pr tbh)

> im thinking we should make a new top level type. `/struct` and we can move that behavior to them while making it seem like less of a eggshell to find out certain datums can't be deleted.

This sort of relies on people understanding "let it live" as a pattern in the first place, which I'm not all that hopeful for. I am sort of tempted to use parent type for this sort of thing but I'm afraid of the pandora's box it'd open. I don't want decl posting yaknow?

Reply:
no need for parent type.

`/struct/callback` will have a parent type of `/datum` by default.